addons/udes_permissions/models/res_groups.py

```diff
@@ -0,0 +1,24 @@
+from odoo import models
+
+
+class ResGroups(models.Model):
+    _inherit = "res.groups"
+
+    def write(self, vals):
+        """Extend write:
+        If we change the name of a group directly linked to a User Template,
+        try to propagate the name change to the User Template
+        """
+        UserTemplate = self.env["user.template"]
+
+        if "name" in vals and vals.get("name"):
+            # @Performance: Ideally we would have a link back directly, but because
+            # the relationship is effectively one2one, we would require a funky hack like this one:
+            # http://blog.odoobiz.com/2014/10/openerp-one2one-relational-field-example.html
+            template_for_group = UserTemplate.search([("template_group_id", "=", self.id)])
```

Comment:
Might be worth considering adding a field to link directly to the template from the group, not a big deal at this early stage though.

Reply:
I did give it a go but it just felt ugly doing the hack specified in the comment. Unless i'm being dumb and forgetting how to do a o2o properly

Reply:
Let's leave it for now then, can always address in the following story if actually needed. :)